### ISSUE TITLE
Module resolution error on NPM and PNPM

### ISSUE THREAD

**Issue by OIRNOIR** [discordjs/discord.js#10018]
### Which package is this bug report for?

discord.js

### Issue description

1. Make a clean install of discord.js using npm or pnpm (This example uses pnpm, but I can reproduce on npm): `pnpm install discord.js`
2. Make any javascript file that requires discord.js. Can literally just be a file with `require('discord.js')`
3. Run the file

You will instantly get the error:
```
node:internal/modules/cjs/loader:598
      throw e;
      ^

Error: Cannot find module '/path/to/your/project/directory/node_modules/.pnpm/@discordjs+rest@2.2.0/node_modules/@vladfrangu/async_event_emitter/dist/index.js'
    at createEsmNotFoundErr (node:internal/modules/cjs/loader:1181:15)
    at finalizeEsmResolution (node:internal/modules/cjs/loader:1169:15)
    at resolveExports (node:internal/modules/cjs/loader:591:14)
    at Module._findPath (node:internal/modules/cjs/loader:668:31)
    at Module._resolveFilename (node:internal/modules/cjs/loader:1130:27)
    at Module._load (node:internal/modules/cjs/loader:985:27)
    at Module.require (node:internal/modules/cjs/loader:1235:19)
    at require (node:internal/modules/helpers:176:18)
    at Object.<anonymous> (/path/to/your/project/directory/node_modules/.pnpm/@discordjs+rest@2.2.0/node_modules/@discordjs/rest/dist/index.js:649:34)
    at Module._compile (node:internal/modules/cjs/loader:1376:14) {
  code: 'MODULE_NOT_FOUND',
  path: '/path/to/your/project/directory/node_modules/.pnpm/@discordjs+rest@2.2.0/node_modules/@vladfrangu/async_event_emitter/package.json'
}
```

This is a critical bug. This completely breaks all new installations of discord.js, and causes a nondiscriminate, ubiquitous crash on startup. Even updating packages puts projects at risk of breaking.

### Code sample

```typescript
require('discord.js')
```

### Versions

- discord.js 14.14.1
- Node v20.10.0
- PNPM 8.11.0 (Reproducible on NPM 10.2.4)
- MacOS Sonoma 14.1.2 (Reproducible on Linux Mint 21.2)

### Issue priority

High (immediate attention needed)

### Which partials do you have configured?

Not applicable

### Which gateway intents are you subscribing to?

Not applicable

### I have tested this issue on a development release

54453b04e

**Comment by OIRNOIR** [comment]
Workaround:
PNPM:
Add to your package.json
```json
"pnpm": {
	"overrides": {
		"@vladfrangu/async_event_emitter": "2.2.2"
	}
}
```

NPM (untested):
Add to your package.json
```json
"overrides": {
	"@vladfrangu/async_event_emitter": "2.2.2"
}
```

**Comment by kai-notifi** [comment]
+1 Seeing this issue too. Looks like an issue with https://github.com/vladfrangu/async_event_emitter

**Comment by NoOneWasTaken** [comment]
As @OIRNOIR stated the workaround for NPM might not work so all you can do is override it directly in package.json.
Follow the simple steps (NPM Users):

Change the following code in node_modules/@vladfrangu/async_event_emitter/package.json:
```
Line 5: "main": "./dist/index.cjs",
Line 18 and 22: "default": "./dist/index.cjs"
```

This should fix the error hupefully :)
[Credits](https://github.com/vladfrangu/async_event_emitter/issues/2#issuecomment-1837366319)

**Comment by aCursedComrade** [comment]
> As @OIRNOIR stated the workaround for NPM might not work so all you can do is override it directly in package.json. Follow the simple steps (NPM Users):
> 
> Change the following code in node_modules/@vladfrangu/async_event_emitter/package.json:
> 
> ```
> Line 5: "main": "./dist/index.cjs",
> Line 18 and 22: "default": "./dist/index.cjs"
> ```
> 
> This should fix the error hupefully :) [Credits](https://github.com/vladfrangu/async_event_emitter/issues/2#issuecomment-1837366319)

This works as temporary fix but it will quickly overridden by npm when you try to manage packages.

I can confirm the fix suggested by @OIRNOIR is working for npm. You need add that override to your `package.json` and do `npm i` for npm to do the changes.

**Comment by NoOneWasTaken** [comment]
> > As @OIRNOIR stated the workaround for NPM might not work so all you can do is override it directly in package.json. Follow the simple steps (NPM Users):
> > Change the following code in node_modules/@vladfrangu/async_event_emitter/package.json:
> > ```
> > Line 5: "main": "./dist/index.cjs",
> > Line 18 and 22: "default": "./dist/index.cjs"
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > This should fix the error hupefully :) [Credits](https://github.com/vladfrangu/async_event_emitter/issues/2#issuecomment-1837366319)
> 
> This works as temporary fix but it will quickly be overridden by npm when you try to manage packages.
> 
> I can confirm the fix suggested by @OIRNOIR is working for npm. You need add that override to your `package.json` and do `npm i` for npm to do the changes.

That works too! I just choose this as I am only using 1 package for the entire project (only discord.js)

**Comment by vladfrangu** [comment]
This has been fixed as of @vladfrangu/async_event_emitter@2.2.4, sorry about that.
